src/NServiceBus.Core/obsoletes.cs

```diff
@@ -344,6 +344,34 @@ public static void RegisterEncryptionService(this EndpointConfiguration config,
             throw new NotImplementedException();
         }
     }
+
+    [ObsoleteEx(
+        Message = "Implement `IWantToRunWhenEndpointStartsAndStops` in the NServiceBus.Host package or NServiceBus.Host.AzureCloudService package instead",
+        RemoveInVersion = "7.0",
+        TreatAsErrorFromVersion = "6.0")]
+    public interface IWantToRunWhenBusStartsAndStops
+    {
+        Task Start(IMessageSession session);
+
+        Task Stop(IMessageSession session);
+    }
+
+    public partial class EndpointConfiguration
+    {
+        /// <summary>
+        /// Register specific instance to run when endpoint starts and stops.
+        /// </summary>
+        /// <param name="callbackObject">Callback object.</param>
+        [ObsoleteEx(
+            Message = "Implement `IWantToRunWhenEndpointStartsAndStops` in the NServiceBus.Host package or NServiceBus.Host.AzureCloudService package instead", 
+            RemoveInVersion = "7.0", 
+            TreatAsErrorFromVersion = "6.0")]
+        public void RunWhenEndpointStartsAndStops(IWantToRunWhenBusStartsAndStops callbackObject)
```

Comment:
Was this one present in v5? (don't look like it https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/BusConfiguration.cs)

If no then we can remove this